### ISSUE TITLE
$mdToast breaks click handlers on hide

### ISSUE THREAD

**Issue by MagicIndustries** [angular/material#3060]
I'm using $mdToast to launch an alert from a PubSub subscription, and I've found an odd problem. I have a screen with an md-list, and when I click an item the app navigates to another screen. If however, I launch a toast alert and wait until it hides itself, clicking on the list item still triggers the ripple effect but nothing else.  I just realised the FAB button on the same page, and all the buttons in the toolbar also stop working. Everything still registers hits visually, but they no longer call the functions they're bound to.

It happens on all the lists in my application. As far as I know I'm not doing anything special to call it. My setup:

a notification directive which listens to a channel.
When a message comes through the channel, it makes this call in the link function:

                       // launch the toast
                        $mdToast.show({
                            scope: scope,
                            controller: 'ToastCtrl',
                            templateUrl: 'components/notifications/toast-template.html',
                            hideDelay: 3000,
                            position: scope.getToastPosition()
                        });

I've tried with my own template and with the default template from the angular-material toast demo, same thing happens either way.

The ToastCtrl looks like this:

    .controller( 'ToastCtrl', function ( $scope, $mdToast ) {
            $scope.closeToast = function () {
                $mdToast.hide();
            };
        });

Inspecting the code after the toast is hidden shows the ng-click still there, but the method it calls is definitely not being invoked. If I click a list item while the toast is showing but before it starts the hide animation, it still works.  

Anyone else seen this and have a quick fix?

**Comment by MagicIndustries** [comment]
I found if I stop passing in a scope and instead use the locals object to inject data into the toast controller,  it stops the problem.  I've no idea what the original problem actually was, but as soon as I stopped sending a scope it stopped.

I can confirm that it happens on hide - both when you click the close button and when you just let it fade. Switching to "locals" resolves the issue for me for right now.
